### DISCUSSION TITLE
conditionally including another qmd file depending on output format

### DISCUSSION THREAD

**Discussion by donboyd5** [quarto-dev/quarto-cli#676] [Q&A] [answered]
I have a project in which I will have html and pdf formats. For the html format it is appropriate to have a section with a dynamic table that the reader can sort, etc. (generated in R with DT::datatable). In the pdf output it does not make sense to include this section and so I would like to exclude it.

It seems like there are two ways in which quarto might have this capability, although I have not found either: 

1. Some sort of rendering-creating recipe that allows the creator of the project to include some qmd files when rendering to one format, and a different set when rendering to a different format. I have looked in the [books](https://quarto.org/docs/books/) documentation to see whether I can specify different chapters for one format than for another, and I experimented with modifying an _quarto.html file to put the chapters in the format sections of the yaml (with different chapters in one format than another) rather than in the book section of _quarto.yaml, but that does not work.

2. Some sort of conditional "include" statement in the main qmd file that includes my dynamic table based on a variable determined at render time. This seems similar to [this discussion](https://community.rstudio.com/t/quarto-book-reference-chunk-data-in-another-file/131967) (but I do not fully understand it), [this discussion](https://github.com/quarto-dev/quarto-cli/discussions/373) (again do not fully understand it) and its [associated pull request](https://github.com/quarto-dev/quarto-cli/pull/386), and it seems to be a little bit related to [this](https://community.rstudio.com/t/quarto-book-reference-chunk-data-in-another-file/131967). However, I have not been able to find a way to include a subdocument, much less include one conditionally. I do see that it is possible to include a qmd child document (a la Rmarkdown) unconditionally, but (a) that doesn't seem like an approach quarto would take across multiple languages, and (b) I have not been able to find a way to do it conditionally based on a variable that I would set (although [this](https://bookdown.org/yihui/rmarkdown-cookbook/child-document.html) suggests it is possible to do it in Rmarkdown) as opposed to quarto.

With that as background, is there a reasonably easy way in quarto to have sections included depending on output format, or to include them conditionally based on variables set at render time? If not, I think it would be a nice feature to add.

Many thanks!

**Comment by jjallaire** [comment, marked as answer]
We are going to add these sorts of capabilities soon (including the ability to optionally embed a static screenshot of JavaScript backed widgets in PDFs Word, etc.). In the meantime, the best approach might be to mark your HTML-specific output with a class and then use a Lua filter to exclude it in non-HTML formats. You can see some format checks in our Lua filters here: https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/filters/common/pandoc.lua. The filter might look something like this (code not tested!)

```lua
function Div(el)
  if el.attr.classes:includes("html-only") and FORMAT == "pdf" then
     return pandoc.Null()
  end
end
```

**Comment by cderv** [comment]
For cross reference, related topic in https://github.com/quarto-dev/quarto-cli/discussions/373
